at_client/lib/src/service/notification_service_impl.dart

```diff
@@ -0,0 +1,236 @@
+import 'dart:async';
+import 'dart:convert';
+import 'package:at_client/at_client.dart';
+import 'package:at_client/src/exception/at_client_exception_util.dart';
+import 'package:at_client/src/manager/monitor.dart';
+import 'package:at_client/src/preference/monitor_preference.dart';
+import 'package:at_client/src/service/connectivity_listener.dart';
+import 'package:at_client/src/service/notification_service.dart';
+import 'package:at_commons/at_commons.dart';
+import 'package:at_utils/at_logger.dart';
+import 'package:at_client/src/responseParser/notification_response_parser.dart';
+
+class NotificationServiceImpl implements NotificationService {
+  Map<String, Function> listeners = {};
+  Map<String, StreamController> streamListeners = {};
+  final EMPTY_REGEX = '';
+  static const notificationIdKey = '_latestNotificationId';
+
+  final _logger = AtSignLogger('NotificationServiceImpl');
+
+  late AtClient atClient;
+
+  bool isMonitorStarted = false;
```

Comment:
Should we make these guys "_" variables or keep it like this.

Reply:
done